### PR TITLE
[15.0][IMP] hr_attendance_report_theoretical_time: Add company compatibility in the report + Add Active employees filter to report

### PR DESCRIPTION
**Pull request by victoralmau**
Changes done:
- [x] Add company compatibility in the report
- [x] Add Active employees filter to report

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT51401

### CODE REVIEW

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ class HrAttendanceTheoreticalTimeReport(models.Model):
     employee_id = fields.Many2one(
         comodel_name="hr.employee", string="Employee", readonly=True
     )
+    company_id = fields.Many2one(related="employee_id.company_id")
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py:23]: Why related if you are selecting it in the query?

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml [hunk 2]
```diff
@@ -65,7 +71,7 @@
         <field name="res_model">hr.attendance.theoretical.time.report</field>
         <field
             name="context"
-        >{'search_default_previous_month': 1, 'search_default_current_month': 1, 'search_default_my': 1}</field>
+        >{'search_default_previous_month': 1, 'search_default_current_month': 1, 'search_default_my': 1, 'search_default_active_employees': 1}</field>
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report_views.xml]: I don't consider this filter should be active by default. You must seen all the attendances, being from archived employees or not.

File: hr_attendance_report_theoretical_time/security/hr_attendance_report_theoretical_time_security.xml [hunk 3]
```diff
@@ -2,6 +2,11 @@
 <!-- Copyright 2018 Tecnativa - Pedro M. Baeza
      License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
 <odoo noupdate="1">
+    <record id="rule_multi_company_theoretical_vs_worked_report" model="ir.rule">
+        <field name="name">Theoretical vs worked hours multi-company</field>
+        <field name="model_id" ref="model_hr_attendance_theoretical_time_report" />
+        <field name="domain_force">[('company_id', 'in', company_ids + [False])]</field>
```
Comment (pedrobaeza) [inline hr_attendance_report_theoretical_time/security/hr_attendance_report_theoretical_time_security.xml]: `company_id` of an employee can be empty?

Comment (victoralmau) [reply hr_attendance_report_theoretical_time/security/hr_attendance_report_theoretical_time_security.xml]: Right, it is a required field, it has been changed.

File: hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py (hunk 1 above)
Comment (victoralmau) [reply hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py:23]: Changed to leave the related field only.

**Review by pedrobaeza — APPROVED** [review]

**Comment by victoralmau** [comment]
Ping @chienandalu

**Review by chienandalu — APPROVED** [review]
:+1:

**Pull request merged**